javascript/javascript1/week2/candy-helper.js

```diff
@@ -0,0 +1,46 @@
+const myCandyBasket = [{ candyType: "sweet", pricePerGram: 0.5 }, { candyType: "toffee", pricePerGram: 1.1 },
+{ candyType: "Chocolate", pricePerGram: 0.7 }, { candyType: "chewingGum", pricePerGram: 0.03 }];
+const boughtCandyPrices = [];
+let amountToSpend = Math.random() * 100;
+let flag = true;
+console.log(`Total money in hand is ${amountToSpend}`);
+console.log(`My candy basket with there type and respective price  pergram is`);
+console.log(myCandyBasket);
+
+function addCandy(name, weight) {
+    const totalPrice = getCandyPrice(name, weight);
+    console.log(`The total price of ${weight}gram ${name} is ${totalPrice}`);
+    if (amountToSpend > totalPrice) {
+        boughtCandyPrices.push({ candyType: name, CandyWeight: weight, CandyTotalPrice: totalPrice });
+        amountToSpend -= totalPrice;
+        console.log(`You have some more to spend for candies.`);
+        return amountToSpend;
+    }
+    else {
+        flag = false;
+        console.log(`You have no more money for candy`);
+    }
+}
+
+
+function getCandyPrice(name, weight) {
```

Comment:
has to in the addCandy function.
According to the homework you have to create two functions, one called add candy another one called canBuyMoreCandy. but It is a little bit confusing here I recommend that you try to rewrite it and try to avoid using unnecessary console.logs .

Reply:
will do,thank you.